src/Stratis.Bitcoin/Configuration/TextFileConfiguration.cs

```diff
@@ -121,13 +122,16 @@ public void MergeInto(TextFileConfiguration destination)
         /// Retrieves all values of a specific argument name (where the name excludes the dash prefix).
         /// </summary>
         /// <param name="key">Name of the argument (excluding the dash prefix).</param>
+        /// <param name="logger">The settings logger used to log the value. Logs on Debug level.</param>
         /// <returns>Values for the specified argument.</returns>
-        public string[] GetAll(string key)
+        public string[] GetAll(string key, ILogger logger = null)
         {
             // Get the values with the - prefix.
             if (!this.args.TryGetValue($"-{key}", out List<string> values))
                 values = new List<string>();
 
+            logger?.LogDebug("{0} entries were returned for the key '{1}'", values.Count, key);
```

Comment:
Perhaps we should actually iterate over the entries and log them?
This will be useful to know for logs.

Reply:
We could do that. I will make the change.

Reply:
thanks